torch/csrc/jit/script/schema_matching.cpp

```diff
@@ -332,6 +332,8 @@ c10::optional<MatchedSchema> tryMatchSchema(
       // Argument has a default value and no value was provided, so use the
       // default
       actual_named_value = NamedValue(*arg.default_value());
+    } else if (arg.type()->cast<OptionalType>()) {
```

Comment:
It seems like this is saying that any optional arguments would implicitly have `None` as a default value? We should be pulling default arguments from the function schema's themselves instead of inserting it automatically.

Reply:
Yeah that's also what I suggested. This should be the schema's responsibility to ensure a None defaults

Reply:
Are you guys saying that when we wrote the schema we should have made sure that all optional arguments had a default argument?

Reply:
Yeah i think so. Thinking of it more, I think the real problem here is that sometimes schema cannot specify the default argument for middle args because C++ cannot have default arg before non-default arg, like `p` in `norm`. Maybe we should do this in schema matching then?

@driazati what do you think?

Reply:
The other args are also missing defaults, at least according to [the docs](https://pytorch.org/docs/stable/torch.html#torch.norm), so it's still at the end and a valid Python schema.

Reply:
sounds good to me. that indeed means schema in native_functions.yaml need to unify to avoid overloads ambiguity for the `norm` and `empty_like` case.